### DISCUSSION TITLE
Opening a file in web socket handler causes socket causes error code 1001

### DISCUSSION THREAD

**Discussion by trzy** [fastapi/fastapi#11273] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request, WebSocket
import uvicorn


app = FastAPI()


@app.websocket("/ws")
async def websocket_endpoint(websocket: WebSocket):
    try:
        fp = open("capture.raw", "wb")
        await websocket.accept()  
        while True:
            data = await websocket.receive_bytes()
            print(f"Message received: {len(data)} bytes")
    except Exception as e:
        print(f"ERROR: {e}")
    


if __name__ == "__main__":
    uvicorn.run(app, host="127.0.0.1", port=8000)
```

### Description

- Send data to this web socket from a browser and observe the error get triggered. The culprit is the file open command. 

### Operating System

macOS

### FastAPI Version

0.110.0

### Pydantic Version

2.6.3

### Python Version

Python 3.11.8

**Comment by YuriiMotov** [comment]
Do you have a traceback of the error from the python console?
I modified your code a bit (closing the file in finally block and catching `WebSocketDisconnect`) and ran this with Pytest. It works without any errors. I use Python 3.11.6 and FastAPI 0.110.0.

```
import pytest
from fastapi import FastAPI, WebSocket, WebSocketDisconnect

app = FastAPI()

@app.websocket("/ws")
async def websocket_endpoint(websocket: WebSocket):
    try:
        fp = open("capture.raw", "wb")
        await websocket.accept()
        while True:
            data = await websocket.receive_bytes()
            print(f"Message received: {len(data)} bytes")
    except WebSocketDisconnect:
        print("Disconnecting")
    except Exception as e:
        print(f"ERROR: {e.__class__},  args {e.args}, line {e.__traceback__.tb_lineno}")
    finally:
        fp.close()

from fastapi.testclient import TestClient

@pytest.mark.asyncio
async def test_websocket():
    client = TestClient(app)
    with client.websocket_connect("/ws") as websocket:
        data = "some data"
        websocket.send_bytes(data.encode())
        data = "some data 2"
        websocket.send_bytes(data.encode())
```
Do you have any errors if you run this code with Pytest?

**Comment by trzy** [comment]
I did more testing. This happens occasionally on Windows too but Windows is more reliable. It may have something to do with the browser side as websocat can connect just fine and send bytes. 

Here is an easier way to test. Use the following HTML file. I think due to browser security policies, this needs to run on another server hosted on your machine (I use the "Live Server" extension in VS Code, which let's you click a button in the lower right of your IDE named 'Go Live' to instantly launch the page from a local http server). 

```
<html>
    <head>
        <script>
                socket = new WebSocket("ws://127.0.0.1:8000/ws");
                socket.onopen = function(event) {
                    console.log("WebSocket connection opened");
                    let utf8Encode = new TextEncoder();
                    let bytes = utf8Encode.encode("hello, world!");
                    socket.send(bytes);
                };
                socket.onerror = function(error) {
                    console.error("WebSocket error:", error);
                };
                socket.onclose = function(event) {
                    console.log("WebSocket connection closed");
                };
        </script>
    </head>
</html>
```

**Comment by trzy** [comment]
No errors if I run the code you pasted with pytest.

Also no errors connecting with websocat. 

This happens only from browsers.

**Comment by trzy** [comment]
Note also, if I serve my small example from the same FastAPI server, it seems to work:

```
@app.get("/", response_class=HTMLResponse)
async def read_items():
    html_content = """
    <html>
    <head>
        <script>
                socket = new WebSocket("ws://127.0.0.1:8000/ws");
                socket.onopen = function(event) {
                    console.log("WebSocket connection opened");
                    let utf8Encode = new TextEncoder();
                    let bytes = utf8Encode.encode("hello, world!");
                    socket.send(bytes);
                };
                socket.onerror = function(error) {
                    console.error("WebSocket error:", error);
                };
                socket.onclose = function(event) {
                    console.log("WebSocket connection closed");
                };
        </script>
    </head>
    </html>
    """
    return HTMLResponse(content=html_content, status_code=200)
    ```
    
    But I was planning to serve the page itself from a different server on the same machine.

**Comment by YuriiMotov** [comment]
I tried to reproduce the error, ran it with "Go live".

I only get this error when I close the web page. And this is totally expected behavior.

I will close this to keep discussions clean. If you still struggle with this, feel free to re-open
